R/plot_forecast.R

```diff
@@ -61,35 +61,19 @@ plot_forecast <- function(forecast_data,
   if(is.na(title))
     stop("title argument interpretable as a character.")
   
-  # optional models and locations
-  if (length(unique(forecast_data$model)) == 1){
-    models = unique(forecast_data$model)
-  } else {
-    if (!missing(models)){
-      if (!all(models %in% forecast_data$model)) {
-        stop("Error in plot_forecast: Not all models are available in forecast data.")
-      }
-    } else {
-      stop("Error in plot_forecast: Please select a model to plot.")
+  # optional models parameter. Default to all models in forecast_data
+  if (!missing(models)){
+    if (!all(models %in% forecast_data$model)) {
+      stop("Error in plot_forecast: Not all models are available in forecast data.")
     }
-  }
-  
-  if (length(unique(forecast_data$location)) == 1){
-    locations = unique(forecast_data$location)
   } else {
-    if (!missing(locations)){
-      if (!all(locations %in% forecast_data$location)) {
-        stop("Error in plot_forecast: Not all locations are available in forecast data.")
-      }
-    } else {
-      stop("Error in plot_forecast: Please select a location to plot.")
-    }
+    models <- unique(forecast_data$model)
   }
   
-  # validate truth_source
-  truth_source <- match.arg(truth_source, 
-                            choices = c("JHU","USAFacts", "NYTimes"), 
-                            several.ok = FALSE)
+  # optional locations parameter. Default to all locations in forecast_data
+  if (missing(locations)){
+    locations <- unique(forecast_data$location)
+  }
```

Comment:
There used to be some validation code like the below.  Do we still have something like that?  (If not, is that ok?)

```
    if (!missing(locations)){
      if (!all(locations %in% forecast_data$location)) {
        stop("Error in plot_forecast: Not all locations are available in forecast data.")
      }
    }
```

Reply:
Yeah location validation is at line 78-94.